.gitignore

```diff
@@ -62,3 +62,12 @@ storybook-static
 ganache-deployments.json
 ganache-deployments*.json
 *.lock
+
+# yarn 2
+# https://yarnpkg.com/advanced/qa#which-files-should-be-gitignored
+.yarn/*
+!.yarn/cache
```

Comment:
It looks like these files are checked in down below.

It would be nice to not have those files as a part of the history. Otherwise, the repo will permanently be `X` megabytes larger.

Reply:
Here is a relevant list of tradeoffs: https://github.com/yarnpkg/berry/issues/180#issuecomment-493789642

I would prefer using `git-lfs` if we choose to do zero-installs, so that we don't end up in this kind of a situation: https://github.com/yarnpkg/berry/issues/180#issuecomment-506022504

Reply:
> Do you intend to check in a cache of each dependency in the repository?

Yes, as per the recommendations for zero-installs.

> I would prefer using git-lfs if we choose to do zero-installs

I looked into this before and I recall that `git-lfs` is not well suited to large numbers of files. It is designed for things like `.psd` files. It is not possible to point `git-lfs` at an entire directory.

Reply:
Then I would advocate for _not_ using zero-installs, and evaluating that choice separately.

Using zero installs has a very high cost: the size of the repo will jump by hundreds of megabytes initially, and by tens to hundreds of megabytes any time we make a large change to our dependencies. This is difficult to change after the fact, as it could require re-writing git history on the `master` branch.

Reply:
> Then I would advocate for _not_ using zero-installs, and evaluating that choice separately.

Noted that zero-installs has disadvantages. I'll try and compile a list of pros and cons specific to our project and we can see whether the tradeoff is acceptable or not.